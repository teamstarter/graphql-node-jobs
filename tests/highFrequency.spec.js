const request = require('supertest')
const addMinutes = require('date-fns/addMinutes')

const {
  migrateDatabase,
  seedDatabase,
  getNewServer,
  models,
  closeEverything,
  deleteTables,
  resetDatabase,
} = require('./test-database.js')
const {
  checkForJobs,
  listJobs,
  getNewClient,
  createJob,
  CancelRequestedError,
} = require('../lib/index')

// This is the maximum amount of time the band of test can run before timing-out
jest.setTimeout(600000)

let server = null
const client = getNewClient(
  `http://localhost:${process.env.PORT || 8080}/graphql`
)

const wait = (time) => new Promise((resolve) => setTimeout(resolve, time))

const acquireJob = (variables) => ({
  query: `mutation($typeList: [String!]!, $workerId: String) {
    acquireJob(
      typeList: $typeList
      workerId: $workerId
    ) {
      id
      name
      status
      output
    }
  }`,
  variables,
  operationName: null,
})

const jobCreate = (variables) => ({
  query: `mutation($job: jobInput!) {
    jobCreate(
      job: $job
    ) {
      id
      name
      status
      output
      jobUniqueId
    }
  }`,
  variables,
  operationName: null,
})

const jobUpdate = (variables) => ({
  query: `mutation($job: jobInput!) {
    jobUpdate(
      job: $job
    ) {
      id
      name
      status
      isUpdateAlreadyCalledWhileCancelRequested
    }
  }`,
  variables,
  operationName: null,
})

const jobList = (variables) => ({
  query: `query($where: SequelizeJSON!) {
    job(
      where: $where
    ) {
      id
      name
      status
      isHighFrequency
    }
  }`,
  variables,
  operationName: null,
})

const customAcquire = (variables) => ({
  query: `mutation($typeList: [String!]!) {
    customAcquire(
      typeList: $typeList
    ) {
      id
    }
  }`,
  variables,
  operationName: null,
})

const recoverJob = (variables) => ({
  query: `mutation($id: Int!) {
    recover(id: $id) {
      id
      status
    }
  }`,
  variables,
  operationName: null,
})

const toggleHoldJobType = (variables) => ({
  query: `mutation($type: String!) {
    toggleHoldJobType(type: $type) {
      id
      type

    }
  }`,
  variables,
  operationName: null,
})

const retryJob = (variables) => ({
  query: `mutation retryJob($id: Int!){
    retryJob(id: $id) {
      id
      name
      type
      status
    }
  }`,
  variables,
  operationName: null,
})

/**
 * Starting the tests
 */
describe('Test the highFrequency field of job entity', () => {
  beforeAll(async () => {
    await migrateDatabase()
    await seedDatabase()
    server = await getNewServer()
  })

  beforeEach(async () => {
    await resetDatabase()
  })

  afterEach(async () => {
    await deleteTables()
  })

  afterAll(async (done) => {
    await closeEverything(server, models, done)
  })

  it('List the jobs with and without highFrequency', async () => {
    const response = await request(server)
      .post('/graphql')
      .send(
        jobList({
          where: { isHighFrequency: false },
        })
      )

    expect(response.body.errors).toBeUndefined()
    expect(response.body.data).toMatchSnapshot()

    const response2 = await request(server)
      .post('/graphql')
      .send(
        jobList({
          where: { isHighFrequency: true },
        })
      )

    expect(response2.body.errors).toBeUndefined()
    expect(response2.body.data).toMatchSnapshot()
  })
})
