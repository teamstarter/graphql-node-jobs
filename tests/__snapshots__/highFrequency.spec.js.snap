// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the highFrequency field of job entity List the jobs with and without highFrequency 1`] = `
Object {
  "job": Array [
    Object {
      "id": 2,
      "isHighFrequency": false,
      "name": "Bla b",
      "status": "queued",
    },
    Object {
      "id": 3,
      "isHighFrequency": false,
      "name": "Bla c",
      "status": "failed",
    },
    Object {
      "id": 4,
      "isHighFrequency": false,
      "name": "Bla d",
      "status": "planned",
    },
    Object {
      "id": 5,
      "isHighFrequency": false,
      "name": "Bla e",
      "status": "cancelled",
    },
  ],
}
`;

exports[`Test the highFrequency field of job entity List the jobs with and without highFrequency 2`] = `
Object {
  "job": Array [
    Object {
      "id": 1,
      "isHighFrequency": true,
      "name": "Bla a",
      "status": "queued",
    },
  ],
}
`;
